common/include/pcl/common/fps_counter.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception Inc.
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/common/time.h>
+
+namespace pcl {
+
+class fpsCounter {
+
+  unsigned tick_;
+  double start_time_;
+
+  struct ret {
+    double fps = 0, duration = 0;
+  };
+
+  void
+  reset()
+  {
+    tick_ = start_time_ = 0;
+  };
+
+  double
+  now() const
+  {
+    return pcl::getTime();
+  };
+
+protected:
+  std::string what_;
+
+public:
+  fpsCounter(const std::string what) : what_(what), tick_(0) {}
+
+  unsigned
+  tick()
+  {
+    if (tick_ == 0) {
+      start_time_ = now();
+    }
+
+    tick_ += 1;
+    if (resetCondition()) {
+      printCounterDetails();
+      reset();
+    }
+
+    return tick_;
+  };
+
+  virtual bool
+  resetCondition()
+  {
+    return getAvgFps().duration >= 1.0;
```

Comment:
Default condition is not good. Perhaps, just a simple `ticks > 1000` for default case